### PR TITLE
Update to Django 3.1

### LINKED ISSUE #2795: Django 3.1
I haven't seen anyone asked this question yet. So I'll go first. Is it ok now to upgrade to django 3.1.x?

Another newbie question: beside the tests that I wrote, is there any other test that I could run to make sure the django-cookiecutter on my computer works with django 3.1.1?

Thank you.

### PR DESCRIPTION
**Pull request by areski**
## Description

Django dependencies upgrade.

Checklist:

- [X] I've made sure that `tests/test_cookiecutter_generation.py` is updated accordingly (especially if adding or updating a template option)
- [X] I've updated the documentation or confirm that my change doesn't require any updates

## Rationale

Fix #2795

### CODE REVIEW

File: tests/test_docker.sh [hunk 1]
```diff
@@ -30,4 +30,4 @@ docker-compose -f local.yml run django pytest
 docker-compose -f local.yml run django python manage.py makemigrations --dry-run --check || { echo "ERROR: there were changes in the models, but migration listed above have not been created and are not saved in version control"; exit 1; }
 
 # Test support for translations
-docker-compose -f local.yml run django python manage.py makemessages
+docker-compose -f local.yml run django python manage.py makemessages --all
```
Comment (areski) [inline tests/test_docker.sh:33]: That was necessary for the test to pass

File: {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/contrib/sites/migrations/0004_alter_options_ordering_domain.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 3.1.6 on 2021-02-04 14:49
```
Comment (areski) [inline {{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/contrib/sites/migrations/0004_alter_options_ordering_domain.py:1]: migration added for test to pass due to:

> docker-compose -f local.yml run django python manage.py makemigrations --dry-run --check || { echo "ERROR: there were changes in the models, but migration listed above have not been created and are not saved in version control"; exit 1; }

File: {{cookiecutter.project_slug}}/requirements/base.txt [hunk 3]
```diff
@@ -29,7 +29,7 @@ uvicorn[standard]==0.13.3  # https://github.com/encode/uvicorn
 
 # Django
 # ------------------------------------------------------------------------------
-django==3.0.11  # pyup: < 3.1  # https://www.djangoproject.com/
+django==3.1.6  # pyup: < 3.1  # https://www.djangoproject.com/
```
Comment (browniebroke) [inline {{cookiecutter.project_slug}}/requirements/base.txt:32]: We also need to update the PyUP filter to keep receiving updates:

```suggestion
django==3.1.6  # pyup: < 3.2  # https://www.djangoproject.com/
```

**Review by browniebroke — COMMENTED** [review]
Code changes look good. We should also update the documentation, there is -at least- the README mentioning Django 3.0:

https://github.com/pydanny/cookiecutter-django/blob/fc01e870bd53d1ab903f99b7fd842054910a8090/README.rst#L43

We'll need to deploy a few basic vanilla projects to production to test.

Thanks for taking this on!

File: {{cookiecutter.project_slug}}/requirements/base.txt (hunk 3 above)
Comment (areski) [reply {{cookiecutter.project_slug}}/requirements/base.txt]: Well spotted!

**Comment by areski** [comment]
Thanks for the review @browniebroke, I have added fixes to the PR.

What's the methodology for making those deployment? Who is making them?
Is it the community running those test manually and fetching this PR to test?

**Comment by browniebroke** [comment]
Needs to be done manually. Can be tested by running the following command and answering the usual questions:

```
cookiecutter https://github.com/areski/cookiecutter-django --checkout fix_django_31
```

**Comment by mmmcorpsvit** [comment]
already released few days ago: https://docs.djangoproject.com/en/dev/releases/3.1.7/

**Comment by areski** [comment]
Thanks @mmmcorpsvit, I just pushed a fix for this.

**Review by browniebroke — APPROVED** [review]
Most packages are offically testing for it, except django-compressor. Done a bit of testing on prod and it seems to be working, with and without it.

**Pull request merged**